code/datums/flock/flock.dm

```diff
@@ -703,6 +686,51 @@ var/flock_signal_unleashed = FALSE
 	if (!flockTurfAllowed(T))
 		return
 
+	if (!flock_conversion_paths)
+		flock_conversion_paths = list()
+
+		for (var/subtype in typesof(/obj/grille))
+			flock_conversion_paths[subtype] = /obj/grille/flock
+		for (var/subtype in typesof(/obj/window))
+			flock_conversion_paths[subtype] = /obj/window/auto/feather
+		for (var/subtype in typesof(/obj/machinery/door))
+			flock_conversion_paths[subtype] = /obj/machinery/door/feather
+		for (var/subtype in typesof(/obj/stool))
+			flock_conversion_paths[subtype] = /obj/stool/chair/comfy/flock
+		for (var/subtype in typesof(/obj/table))
+			flock_conversion_paths[subtype] = /obj/table/flock/auto
+		for (var/subtype in typesof(/obj/machinery/light))
+			flock_conversion_paths[subtype] = /obj/machinery/light/flock
+		for (var/subtype in typesof(/obj/machinery/light/small/floor))
+			flock_conversion_paths[subtype] = /obj/machinery/light/flock/floor
+		for (var/subtype in typesof(/obj/storage/closet))
+			flock_conversion_paths[subtype] = /obj/storage/closet/flock
+		for (var/subtype in typesof(/obj/storage/secure/closet))
+			flock_conversion_paths[subtype] = /obj/storage/closet/flock
+		for (var/subtype in typesof(/obj/machinery/computer))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute
+		for (var/subtype in typesof(/obj/machinery/computer3))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute
+		for (var/subtype in typesof(/obj/machinery/networked/teleconsole))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute
+		for (var/subtype in typesof(/obj/machinery/networked/mainframe))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute/mainframe
+
+		var/types_to_skip = list(/obj/machinery/computer/card/portable, /obj/machinery/computer/security/wooden_tv,
```

Comment:
this is better, but I still think it would make sense as a check than maintaining a list of hundreds of subtypes